File: app/src/main/java/com/websarva/wings/android/zuboradiary/data/network/GeoCoordinates.java
package com.websarva.wings.android.zuboradiary.data.network;

import androidx.annotation.FloatRange;

public class GeoCoordinates {

    @FloatRange(from = -90.0, to = 90.0)
    private final double latitude;
    @FloatRange(from = -180.0, to = 180.0)
    private final double longitude;

    public GeoCoordinates(double latitude, double longitude) {
        if (latitude < -90) throw new IllegalArgumentException();
        if (latitude > 90) throw new IllegalArgumentException();
        if (longitude < -180) throw new IllegalArgumentException();
        if (longitude > 180) throw new IllegalArgumentException();

        this.latitude = latitude;
        this.longitude = longitude;
    }

    public double getLatitude() {
        return latitude;
    }

    public double getLongitude() {
        return longitude;
    }
}

File: app/src/main/java/com/websarva/wings/android/zuboradiary/data/model/GeoCoordinates.kt
package com.websarva.wings.android.zuboradiary.data.model

import androidx.annotation.FloatRange

class GeoCoordinates(latitude: Double, longitude: Double) {

    @FloatRange(from = -90.0, to = 90.0)
    val latitude: Double
    @FloatRange(from = -180.0, to = 180.0)
    val longitude: Double

    init {
        require(latitude >= -90)
        require(latitude <= 90)
        require(longitude >= -180)
        require(longitude <= 180)

        this.latitude = latitude
        this.longitude = longitude
    }
}
